src/main/java/dev/jbang/EditorManager.java

```diff
@@ -0,0 +1,61 @@
+package dev.jbang;
+
+import static dev.jbang.cli.BaseCommand.EXIT_UNEXPECTED_STATE;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class EditorManager {
+	private static final String CODIUM_DOWNLOAD_URL = "https://github.com/VSCodium/vscodium/releases/download/%s/VSCodium-%s-%s-%s.zip";
+
+	private static Path getEditorPath() {
+		return Settings.getCacheDir(Settings.CacheClass.editors);
+	}
+
+	public static Path downloadAndInstallEditor() {
+		// https://github.com/VSCodium/vscodium/releases/download/1.52.1/VSCodium-darwin-x64-1.52.1.zip
+		String version = "1.52.1";
+
+		Util.infoMsg("Downloading VSCodium " + version
+				+ ". Be patient, this can take several minutes...(Ctrl+C if you want to cancel)");
+		String os = Util.getOS().name().equals("mac") ? "darwin" : Util.getOS().name();
+		String url = String.format(CODIUM_DOWNLOAD_URL, version, os, Util.getArch().name(), version);
+		Util.verboseMsg("Downloading " + url);
+		Path editorDir = getEditorPath();
+		Path editorTmpDir = editorDir.getParent().resolve(editorDir.getFileName().toString() + ".tmp");
```

Comment:
I think it becomes time to create a standard "create and switch" system. We must have implemented this old/tmp/move-dance at least three times now :-)

Reply:
https://github.com/jbangdev/jbang/issues/640